hostpolicy/api/permissions.py

```diff
@@ -13,15 +16,58 @@ def is_super_or_hostpolicy_admin(user):
 
 class IsSuperOrHostPolicyAdminOrReadOnly(IsAuthenticated):
     """
-    Permit user if in super or group admin group, else read only.
+    Permit user if in super or group admin group, or has been granted access through a
+    NetGroupRegexPermission, else read only.
     """
 
     def has_permission(self, request, view):
         if not super().has_permission(request, view):
+            # Not even reading is allowed if you're not authenticated
             return False
         if request.method in SAFE_METHODS:
             return True
-        return is_super_or_hostpolicy_admin(request.user)
+        if is_super_or_hostpolicy_admin(request.user):
+            return True
+
+        # Is this request about atoms or something else that isn't a role?
+        # In that case, non-admin-users shouldn't have access anyway, and we can deny the request.
+        if not (view.__class__.__name__ == 'HostPolicyRoleHostsDetail' or
+                view.__class__.__name__ == 'HostPolicyRoleHostsList'):
+            return False
+
+        # Find out which labels are attached to this role
+        role = HostPolicyRole.objects.get(name=view.kwargs['name'])
+        role_labels = []
+        for label in role.labels.all():
+            role_labels.append(label.name)
+
+        # Find all the NetGroupRegexPermission objects that correspond with
+        # the ipaddress, hostname, and the groups that the user is a member of
+        if 'host' in view.kwargs:
+            hostname = view.kwargs['host']
+        else:
+            hostname = request.data.get("name")
+        host = Host.objects.get(name=hostname)
+        ips = []
+        hostserr = HostSerializer(host)
+        for i in hostserr.data['ipaddresses']:
+            ips.append(i['ipaddress'])
```

Comment:
``ips = Host.objects.filter(name=hostname).values_list('ipaddresses__ipaddress', flat=True)``

Reply:
Had to filter out some `None` values, ended up with this:
https://github.com/unioslo/mreg/blob/17eb02178d5f4025b27754ab3ce4def75e06fd8e/hostpolicy/api/permissions.py#L47